fmriprep/interfaces/multiecho.py

```diff
@@ -78,320 +74,44 @@ def _run_interface(self, runtime):
         return runtime
 
 
-class MaskT2SMapInputSpec(BaseInterfaceInputSpec):
-    image = File(mandatory=True, exists=True, desc='T2* volume to mask')
-    mask = File(mandatory=True, exists=True,
-                desc='skull-stripped mean optimal combination volume')
-    compress = traits.Bool(True, usedefault=True,
-                           desc='use gzip compression on .nii output')
-
-
-class MaskT2SMapOutputSpec(TraitedSpec):
-    masked_t2s = File(exists=True, desc='masked T2* map')
-
-
-class MaskT2SMap(SimpleInterface):
-    """
-    Masks an existing T2* map using the skull-stripped optimally combined
-    volume (i.e., a weighted combination of multi-echo data).
-
-    Example
-    =======
-    >>> from fmriprep.interfaces import multiecho
-    >>> mask_t2s = multiecho.MaskT2SMap()
-    >>> mask_t2s.inputs.image = 'sub-01_run-01_t2smap.nii.gz'
-    >>> mask_t2s.inputs.mask = 'sub-01_run-01_optcomb.nii.gz'
-    >>> res = mask_t2s.run() # doctest: +SKIP
-    """
-
-    input_spec = MaskT2SMapInputSpec
-    output_spec = MaskT2SMapOutputSpec
-
-    def _run_interface(self, runtime):
-        ext = '.nii.gz' if self.inputs.compress else '.nii'
-        flat_mask = apply_mask(self.inputs.image, self.inputs.mask)
-        masked_image = unmask(flat_mask, self.inputs.mask)
-        self._results['masked_t2s'] = fname_presuffix(
-            self.inputs.image, suffix='_masked' + ext, newpath=runtime.cwd, use_ext=False)
-        masked_image.to_filename(self._results['masked_t2s'])
-        return runtime
-
-
 class T2SMapInputSpec(BaseInterfaceInputSpec):
-    in_files = InputMultiPath(File(exists=True), mandatory=True,
+    in_files = InputMultiPath(File(exists=True), mandatory=True, minlen=2,
                               desc='multi-echo BOLD EPIs')
     te_list = traits.List(traits.Float, mandatory=True, desc='echo times')
-    compress = traits.Bool(True, usedefault=True,
-                           desc='use gzip compression on .nii output')
 
 
 class T2SMapOutputSpec(TraitedSpec):
-    t2s_vol = File(exists=True, desc='T2* map')
-    opt_comb = File(exists=True, desc='optimal combination of echos')
+    t2sv = File(exists=True, desc='limited T2* map')
+    s0v = File(exists=True, desc='limited s0 map')
+    t2svG = File(exists=True, desc='adaptive T2* map')
+    s0vG = File(exists=True, desc='adaptive s0 map')
+    ts_OC = File(exists=True, desc='optimally combined ME-EPI time series')
```

Comment:
I would prefer more descriptive output names, such as `t2star_map`, `s0_map`, etc.

Reply:
Yes, I can definitely do that ! Sorry, these were taken from tedana itself.